### PR TITLE
Fixing several warnings, const/unconst and perm_vector related

### PR DESCRIPTION
**Pull request by arielalmendral**

### CODE REVIEW

File: devel/libenkf/src/enkf_main.c [hunk 1]
```diff
@@ -1320,7 +1320,7 @@ bool enkf_main_smoother_update(enkf_main_type * enkf_main , enkf_fs_type * sourc
 
 
 
-
+#ifdef QUEUE_FAILURE_CODE
```
Comment (joakim-hove) [inline devel/libenkf/src/enkf_main.c]: What is this `#ifdef`?

**Comment by arielalmendral** [comment]
The code is not in use...

**Comment by joakim-hove** [comment]
> The code is not in use...

OK - then we remove it.

**Comment by joakim-hove** [comment]
I'll merge when the code which had been `#ifdef`'ed out is removed.

**Comment by arielalmendral** [comment]
Unused code removed.

**Comment by joakim-hove** [comment]
Thank you - in she goes.

**Pull request merged**